sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/orc/OrcScanBuilder.scala

```diff
@@ -60,10 +61,8 @@ case class OrcScanBuilder(
         // changed `hadoopConf` in executors.
         OrcInputFormat.setSearchArgument(hadoopConf, f, schema.fieldNames)
       }
-      val dataTypeMap = schema.map(f => quoteIfNeeded(f.name) -> f.dataType).toMap
-      // TODO (SPARK-25557): ORC doesn't support nested predicate pushdown, so they are removed.
-      val newFilters = filters.filter(!_.containsNestedColumn)
-      _pushedFilters = OrcFilters.convertibleFilters(schema, dataTypeMap, newFilters).toArray
```

Comment:
Just a question. Is it safe to remove this from `DSv2` code path when the configuration still has `effective with file-based data sources in DSv1.`?

Reply:
The config is for DSv1 compatibility issues, and so only controls DSv1 file-based data sources. For DSv2, it is up to the DS implementation. Once the interface is implemented, we think it supports nested filter pushdown. Our orc filter tests cover both v1 and v2.

Reply:
For Orc, there is a `spark.sql.orc.filterPushdown` which can control push down behavior.

Reply:
Got it. Thanks!